### ISSUE TITLE
last release 1.9.4 isn't working on RHEL8 launching using AWX, 1.9.3 is working fine

### ISSUE THREAD

**Issue by arequipeno** [containers/ansible-podman-collections#454]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

ansible-2.9.23-1.el7.noarch
rh-python36-ansible-2.9.6-1.noarch
rh-python36-ansible-runner-1.4.4-1.noarch
rh-python36-ansible-tower-cli-3.3.9-1.noarch
ansible-awx-9.2.0.164-1.el7.x86_64
python3-podman-4.0.0-1.module+el8.6.0+14877+f643d2d6.noarch
podman-4.0.2-6.module+el8.6.0+14877+f643d2d6.x86_64
podman-catatonit-4.0.2-6.module+el8.6.0+14877+f643d2d6.x86_64

**Description**

**Steps to reproduce the issue:**

1. code have requirements.yml to load collection this is working

2. at the execution of code all previous step of deployment working fine and actions using collection failed after login

**Describe the results you received:**
on 1.9.4 latest release
TASK [Podman registry login] ***************************************************
changed: [toto]
TASK [Pause 10 min] ************************************************************
Pausing for 600 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
ok: [toto]
TASK [Pull an image] ***********************************************************
fatal: [toto]: FAILED! => {"msg": "Could not find imported module support code for common.  Looked for either LooseVersion.py or version.py"}

**Describe the results you expected:**
on 1.9.3

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version of the `containers.podman` collection:**
**Either git commit if installed from git: `git show --summary`**
**Or version from `ansible-galaxy` if installed from galaxy: `ansible-galaxy collection list | grep containers.podman`**

```
install by requirements.yml
```

**Output of `ansible --version`:**
```
ansible 2.9.23
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Aug 13 2020, 02:51:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

```

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.0.2
API Version:  4.0.2
Go Version:   go1.17.7

Built:      Tue Apr 19 12:16:32 2022
OS/Arch:    linux/amd64

```

**Output of `podman info --debug`:**

``` yaml
host:
  arch: amd64
  buildahVersion: 1.24.1
  cgroupControllers:
  - cpuset
  - cpu
  - cpuacct
  - blkio
  - memory
  - devices
  - freezer
  - net_cls
  - perf_event
  - net_prio
  - hugetlb
  - pids
  - rdma
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.0-1.module+el8.6.0+14877+f643d2d6.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: 87b7a9037cbd1d81288bdf2d6705dfda889f7cf9'
  cpus: 2
  distribution:
    distribution: '"rhel"'
    version: "8.6"
  eventLogger: file
  hostname: toto
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 4.18.0-372.13.1.el8_6.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 314155008
  memTotal: 3796709376
  networkBackend: cni
  ociRuntime:
    name: runc
    package: runc-1.0.3-2.module+el8.6.0+14877+f643d2d6.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.3
      spec: 1.0.2-dev
      go: go1.17.7
      libseccomp: 2.5.2
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.8-2.module+el8.6.0+14877+f643d2d6.x86_64
    version: |-
      slirp4netns version 1.1.8
      commit: d361001f495417b880f20329121e3aa431a8f90f
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 8485117952
  swapTotal: 8589930496
  uptime: 235h 5m 57.44s (Approximately 9.79 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 1
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 5
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.0.2
  Built: 1650363392
  BuiltTime: Tue Apr 19 12:16:32 2022
  GitCommit: ""
  GoVersion: go1.17.7
  OsArch: linux/amd64
  Version: 4.0.2

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
python3-podman-4.0.0-1.module+el8.6.0+14877+f643d2d6.noarch
podman-4.0.2-6.module+el8.6.0+14877+f643d2d6.x86_64
podman-catatonit-4.0.2-6.module+el8.6.0+14877+f643d2d6.x86_64
```

**Playbook you run with ansible (e.g. content of `playbook.yaml`):**

``` yaml
  - name: Podman registry login
    podman_login:
      username: '{{ DOCKER_USER }}'
      password: '{{ DOCKER_PASSWORD }}'
      registry: '{{ DOCKER_REGISTRY_HOST }}'

  - name: Pull an image
    podman_image:
      name: '{{ DOCKER_REGISTRY_HOST }}/{{ DOCKER_REGISTRY_PROJECT }}/toto:${DOCKER_IMAGE_TAG}'

  - name: Run Web application container
    podman_container:
      name: toto
      image: '{{ DOCKER_REGISTRY_HOST }}/{{ DOCKER_REGISTRY_PROJECT }}/toto:${DOCKER_IMAGE_TAG}'
      state: started
      restart_policy: 'always'
      ports:
        - '{{ EXTERNAL_PORT }}:4000'
      volume:
        - '{{ APPLICATION_PATH }}/krb5.conf:/etc/krb5.conf:ro'
        - '{{ APPLICATION_PATH }}/app.keytab:/etc/krb5.keytab:ro'
        - /etc/pki/ca-trust/source/anchors:/usr/local/share/host-ca-certificates:ro
      env_file: '{{ APPLICATION_PATH }}/.env'
      log_driver: journald
      log_opt:
        tag: toto

  - name: Podman registy logout
    podman_logout:
      registry: '{{ DOCKER_REGISTRY_HOST }}'

```

**Command line and output of ansible run with high verbosity**

**Please NOTE: if you submit a bug about idempotency, run the playbook with `--diff` option, like:**

`ansible-playbook -i inventory --diff -vv playbook.yml`

```
N/A only play on one server
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by arequipeno** [comment]
Thnks in advance

**Comment by sshnaidm** [comment]
probably it's because of https://github.com/containers/ansible-podman-collections/commit/2577f5b0f516dd0657a91a722bac4dc810803062
Which python version do you use on the controller (host that executes Ansible)?

**Comment by sshnaidm** [comment]
@arequipeno any progress with it?

**Comment by arequipeno** [comment]
python 3.6

**Comment by thekad** [comment]
some more info, this happens with a Centos 8 Stream image + `centos-release-ansible-29-1-2.el8s.noarch` + `ansible-2.9.12-1.el8.noarch` (due to several reasons, we can't jump to ansible 2.12+ yet) here are the versions:

```bash
$ cat /etc/os-release 
NAME="CentOS Stream"
VERSION="8"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Stream 8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"

$ ansible --version
ansible 2.9.12
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/dci/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.8 (default, Oct 26 2022, 09:13:21) [GCC 8.5.0 20210514 (Red Hat 8.5.0-17)]
```

Running with collection v1.9.3 works:

```bash
$ grep version /usr/share/ansible/collections/ansible_collections/containers/podman/MANIFEST.json
  "version": "1.9.3",
$ ansible -m containers.podman.podman_container -a 'name=foo image=alpine:latest' localhost -b
localhost | CHANGED => {            
    "actions": [                
        "started foo"                 
    ],                                           
    "changed": true,                   
...
```

But v1.9.4 fails

```bash
$ grep version /usr/share/ansible/collections/ansible_collections/containers/podman/MANIFEST.json
  "version": "1.9.4",
$ ansible -m containers.podman.podman_container -a 'name=foo image=alpine:latest' localhost -b -vvv
ansible 2.9.12
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/dci/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.8 (default, Oct 26 2022, 09:13:21) [GCC 8.5.0 20210514 (Red Hat 8.5.0-17)]
Using /etc/ansible/ansible.cfg as config file
host_list declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
script declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Parsed /etc/ansible/hosts inventory source with ini plugin
META: ran handlers
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: dci
<127.0.0.1> EXEC /bin/sh -c 'echo ~dci && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/dci/.ansible/tmp `"&& mkdir "` echo /home/dci/.ansible/tmp/ansible-tmp-1669762002.6824803-27490-50873305003160 `" && echo ansible-tmp-1669762002.6824803-27490-50873305003160="` echo /home/dci/.ansible/tmp/ansible-tmp-1669762002.6824803-27490-50873305003160 `" ) && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/dci/.ansible/tmp/ansible-tmp-1669762002.6824803-27490-50873305003160/ > /dev/null 2>&1 && sleep 0'
localhost | FAILED! => {
    "msg": "Could not find imported module support code for common.  Looked for either LooseVersion.py or version.py"
}
```

**Comment by sergei-maertens** [comment]
We are running into the same error as above with Python 3.6.8, Ansible 2.9.11 (yes, we need to upgrade our deps) on RHEL 8.

Using Collection version 1.9.3 (instead of 1.9.4) works for us for now!

**Comment by arequipeno** [comment]
Sorry for the delay, I already solved the issue using python venv based on 3.8 and 3.9 since month now
